#### Using scikit-learn to fit a classifier

* All machine learning models in scikit-learn are implemented as Python Classes.
    * They implement the algorithm for learning and predicting.
    * Store the info learned from the data
    
* Training a model on the data = "fitting" a model to the data
    * use .fit() method
 
* To predict the labels of new data: predict() method

#### Here we will use k- nearest neighbor classifier on iris dataser|

In [1]:
from sklearn.neighbors import KNeighborsClassifier

# import iris dataser
from sklearn import datasets
iris = datasets.load_iris()

In [2]:
knn = KNeighborsClassifier(n_neighbors = 6)

* Two arguments are passed: features as a numpy array, and the label also as a numpy array
* Scikit library requires you to have data as a numpy array or pandas dataframe
* Features should take on continuous values
* There should not be any missing values in the dataset

In [3]:
knn.fit(iris['data'], iris['target'])

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=1, n_neighbors=6, p=2,
           weights='uniform')

In [4]:
iris['data'].shape

(150, 4)

* Looking at iris dataset we see that there are 150 observations and 4 features.
* Similary the target needs to be a single column with the same number of observations.

In [5]:
iris['target'].shape

(150,)

In [6]:
# To Do:

# prediction = knn.predict(X_new)
# X_new.shape()
# Output : (3,4)
# print('Prediction {}'.format(prediction
# Output: Prediction [1 1 0]
# 1 - versicolor, 0 - Sertosa